zerver/models.py

```diff
@@ -2145,17 +2146,26 @@ class PreregistrationUser(models.Model):
 
 def filter_to_valid_prereg_users(
     query: QuerySet,
-    invite_expires_in_days: Optional[int] = None,
+    invite_expires_in_days: Union[Optional[int], NoValueSent] = NoValueSent(),
```

Comment:
Can you add a docstring explaining what the `invite_expires_in_days` parameter here means?  I don't think I understand that, and am having a hard time parsing this function.

Reply:
It basically helps us from preventing sending the email if `invite_expires_in_days` number of days has passed since creating the invitation. For eg - I invited a user with expiration time set to 3 days on a particualr day, but the queue processor tries to send the email after 3 days. I am not sure whether this will actually happen.

Reply:
OK. I think this logic is just wrong and we should delete support for that parameter.  AFAIK the only place we call this is here:
```
        invitee = filter_to_valid_prereg_users(                                                                                                          
            PreregistrationUser.objects.filter(id=data["prereg_id"]), invite_expires_in_days                                                             
        ).first()                                                                                                                                        
        if invitee is None:                                                                                                                              
            # The invitation could have been revoked                                                                                                     
            return                                                                                                                                       
```

and as one can see, we already have the `id` for the `PreregistrationUser` object, so we should just look it up by ID and then check directly whether it's expired/revoked, and if so, remove it.

@mateuszmandera can you help confirm my understanding here?  We should fix this in a follow-up since it's a preexisting issue, but I'd like to make sure it doesn't get lost.